### PR TITLE
snap: vendor flutter app in, run standalone (`core22`)

### PR DESCRIPTION
**Pull request by Saviq**

### CODE REVIEW

File: snap/snapcraft.yaml [hunk 1]
```diff
@@ -112,6 +125,10 @@ plugs:
 slots:
   wayland:
 
+slots:
+  wayslot:
+    interface: wayland
```
Comment (mattkae) [inline snap/snapcraft.yaml]: Duplicated key with above^?

**Review by mattkae — CHANGES_REQUESTED** [review]
When removing the duplicated key, I get:

```
2024-05-23 09:14:57.895 :: 2024-05-23 09:14:53.342 craft_parts.errors.PartFilesConflict: Failed to stage: parts list the same file with different contents or permissions.
2024-05-23 09:14:57.895 :: 2024-05-23 09:14:53.342 Parts 'mir-flutter-app' and 'gtk3app' list the following files, but with different contents or permissions:
2024-05-23 09:14:57.895 :: 2024-05-23 09:14:53.342     usr/bin/gtk-update-icon-cache
2024-05-23 09:14:57.895 :: 2024-05-23 09:14:53.342     usr/lib/x86_64-linux-gnu/gtk-3.0/3.0.0/immodules/im-am-et.so
2024-05-23 09:14:57.895 :: 2024-05-23 09:14:53.342     usr/lib/x86_64-linux-gnu/gtk-3.0/3.0.0/immodules/im-broadway.so
2024-05-23 09:14:57.895 :: 2024-05-23 09:14:53.342     usr/lib/x86_64-linux-gnu/gtk-3.0/3.0.0/immodules/im-cedilla.so
2024-05-23 09:14:57.895 :: 2024-05-23 09:14:53.342     usr/lib/x86_64-linux-gnu/gtk-3.0/3.0.0/immodules/im-cyrillic-translit.so
2024-05-23 09:14:57.895 :: 2024-05-23 09:14:53.342     usr/lib/x86_64-linux-gnu/gtk-3.0/3.0.0/immodules/im-inuktitut.so
2024-05-23 09:14:57.895 :: 2024-05-23 09:14:53.343     usr/lib/x86_64-linux-gnu/gtk-3.0/3.0.0/immodules/im-ipa.so
2024-05-23 09:14:57.895 :: 2024-05-23 09:14:53.343     usr/lib/x86_64-linux-gnu/gtk-3.0/3.0.0/immodules/im-multipress.so
2024-05-23 09:14:57.895 :: 2024-05-23 09:14:53.343     usr/lib/x86_64-linux-gnu/gtk-3.0/3.0.0/immodules/im-thai.so
2024-05-23 09:14:57.895 :: 2024-05-23 09:14:53.343     usr/lib/x86_64-linux-gnu/gtk-3.0/3.0.0/immodules/im-ti-er.so
2024-05-23 09:14:57.895 :: 2024-05-23 09:14:53.343     usr/lib/x86_64-linux-gnu/gtk-3.0/3.0.0/immodules/im-ti-et.so
2024-05-23 09:14:57.895 :: 2024-05-23 09:14:53.343     usr/lib/x86_64-linux-gnu/gtk-3.0/3.0.0/immodules/im-viqr.so
2024-05-23 09:14:57.895 :: 2024-05-23 09:14:53.343     usr/lib/x86_64-linux-gnu/gtk-3.0/3.0.0/immodules/im-wayland.so
2024-05-23 09:14:57.895 :: 2024-05-23 09:14:53.343     usr/lib/x86_64-linux-gnu/gtk-3.0/3.0.0/immodules/im-waylandgtk.so
2024-05-23 09:14:57.895 :: 2024-05-23 09:14:53.343     usr/lib/x86_64-linux-gnu/gtk-3.0/3.0.0/immodules/im-xim.so
2024-05-23 09:14:57.895 :: 2024-05-23 09:14:53.343     usr/lib/x86_64-linux-gnu/gtk-3.0/3.0.0/printbackends/libprintbackend-cups.so
2024-05-23 09:14:57.895 :: 2024-05-23 09:14:53.343     usr/lib/x86_64-linux-gnu/gtk-3.0/3.0.0/printbackends/libprintbackend-file.so
2024-05-23 09:14:57.895 :: 2024-05-23 09:14:53.343     usr/lib/x86_64-linux-gnu/gtk-3.0/3.0.0/printbackends/libprintbackend-lpr.so
2024-05-23 09:14:57.895 :: 2024-05-23 09:14:53.343     usr/lib/x86_64-linux-gnu/gtk-3.0/3.0.0/printbackends/libprintbackend-test.so
2024-05-23 09:14:57.895 :: 2024-05-23 09:14:53.343     usr/lib/x86_64-linux-gnu/libgdk-3.so.0.2404.29
2024-05-23 09:14:57.895 :: 2024-05-23 09:14:53.343     usr/lib/x86_64-linux-gnu/libgtk-3-0/gtk-query-immodules-3.0
2024-05-23 09:14:57.895 :: 2024-05-23 09:14:53.343     usr/lib/x86_64-linux-gnu/libgtk-3.so.0.2404.29
2024-05-23 09:14:57.895 :: 2024-05-23 09:14:53.343     usr/share/doc/gtk-update-icon-cache/changelog.Debian.gz
2024-05-23 09:14:57.895 :: 2024-05-23 09:14:53.343     usr/share/doc/libgtk-3-0/changelog.Debian.gz
2024-05-23 09:14:57.895 :: 2024-05-23 09:14:53.343     usr/share/doc/libgtk-3-common/changelog.Debian.gz
```

File: snap/snapcraft.yaml (hunk 1 above)
Comment (Saviq) [reply snap/snapcraft.yaml]: Thanks, fixed.

**Comment by Saviq** [comment]
> When removing the duplicated key, I get:
> ```
> Parts 'mir-flutter-app' and 'gtk3app' list the following files, but with different contents or permissions:
> ```

That's weird… tried after `snapcraft clean`? Let's see what CI says…

**Comment by Saviq** [comment]
> That's weird… tried after `snapcraft clean`? Let's see what CI says…

Yeah must've been a stale `gtk3app`. `snapcraft clean gtk3app` should be enough.

**Comment by mattkae** [comment]
> > That's weird… tried after `snapcraft clean`? Let's see what CI says…
> 
> Yeah must've been a stale `gtk3app`. `snapcraft clean gtk3app` should be enough.

Yup that was my issue indeed!

**Comment by Saviq** [comment]
The failures now are:

https://bugs.launchpad.net/review-tools/+bug/2066923

**Review by mattkae — APPROVED** [review]
Good by me :ok_hand:

**Review by mattkae — CHANGES_REQUESTED** [review]
Hm I can't seem to run things like `qt-test`. Are you having any better luck:

```
ERROR: ./src/server/graphics/default_configuration.cpp(198): Throw in function virtual const std::vector<std::shared_ptr<mir::graphics::DisplayPlatform> >& mir::DefaultServerConfiguration::the_display_platforms()
Dynamic exception type: boost::wrapexcept<std::runtime_error>
std::exception::what: Exception while creating graphics platform
ERROR: ./src/server/graphics/platform_probe.cpp(232): Throw in function std::vector<std::pair<mir::graphics::SupportedDevice, std::shared_ptr<mir::SharedLibrary> > > mir::graphics::modules_for_device(const std::function<std::vector<mir::graphics::SupportedDevice>(const mir::SharedLibrary&)>&, const std::vector<std::shared_ptr<mir::SharedLibrary> >&, mir::graphics::TypePreference)
Dynamic exception type: boost::wrapexcept<std::runtime_error>
std::exception::what: Failed to find any platforms for current system

```

**Comment by Saviq** [comment]
> Hm I can't seem to run things like `qt-test`. Are you having any better luck:

Did you pull the latest changes?

**Review by mattkae — APPROVED** [review]
Working for me after latest update!

**Pull request merged**